### DISCUSSION TITLE
FormatConverterStream : setAudioInfo() does only update the sampleRateConverter?

### DISCUSSION THREAD

**Discussion by mirkosertic** [pschatzmann/arduino-audio-tools#1718] [Q&A] [answered]
Hi!

Maybe I got something wrong, but I've noted that the following code sample results in wrong audio conversation with the FormatConverter class:

```
formatconverter.begin(AudioInfo(16000, 2, 16));
formatconverter.setAudioInfo(AudioInfo(16000, 1, 16));
``` 

The FormatConverter contains correct AudioInfo on input and output side, but the setAudioInfo() implementation seems to only update the internal sampleRateConverter, but not the in begin() initialized numberFormatConverter or channelFormatConverter. So it effectively seems to ignore the stereo to mono conversion.

Is this by design? I need to use the FormatConverter as a post processor for decoded MP3 data, and AudioInfo propagation works correctly in the pipeline, but the described internal data structures might not be handled properly.

Thank you for your support!

Mirko

**Comment by pschatzmann** [comment, marked as answer]
This seems to be a bug. I committed a correction

**Reply by mirkosertic** [reply]
Thank you :-)
